refactor: Create 'operator' folder (move docs)

Roming22 opened the pull request:
This commit is part of a larger set of commits with the goal of restructuring the repository.

Signed-off-by: Romain Arnaud <rarnaud@redhat.com>

bnallapeta reviewed (commented):
- I believe kcp-registration.md and operations.md are now moved to Architect folder. Should be deleted from here.
- Git url in [this](https://github.com/Roming22/pipeline-service/blob/operator-docs/operator/docs/sre/examples/single_cluster/environment/compute/mycluster-com/kustomization.yaml) and [this](https://github.com/Roming22/pipeline-service/blob/operator-docs/operator/docs/sre/examples/single_cluster/environment/kcp/kustomization.yaml) needs to be updated.

Roming22 commented:
@bnallapeta 

> * I believe kcp-registration.md and operations.md are now moved to Architect folder. Should be deleted from here.

The diff tells me that the file are move from the `architect` folder. My understanding is that it is OK.

> * Git url ... 

Addressed by your edit.

bnallapeta commented:
@Roming22 Ack. Merging the PR.
